CHANGELOG.md

```diff
@@ -4,6 +4,37 @@
 `psql` with the `-X` flag to prevent any `.psqlrc` commands from
 accidentally triggering the load of a previous DB version.**
 
+
+## 2.16.0 (2024-07-18)
+
+This release contains performance improvements and bug fixes since
+the 2.15.3 release. We recommend that you upgrade at the next
+available opportunity.
+
```

Comment:
```suggestion

With this release we introduce a lot of performance focused optimizations, both for DML over compressed chunks, 
but also for speeding up those tricky queries that end up accessing more chunks than needed, by enabling chunk
exclusion using additional, non partitioning columns.
We also introduce new options for Timescale users that like their Foreign Keys, allowing for FKs from regular tables towards hypertables, but also removing some really annoying locks in the reverse direction that blocked access to referenced tables while compression was running. 
Finally, big updates on the Continuous Aggregates front - more types of joins supported, additional equality operators on join clauses and support for joining with multiple regular tables. 

**Highlighted features in this release**
* Improved query performance through chunk exclusion using additional columns:
You can now define sparse indexes on chunks for any column with
an integer data type (smallint, int, bigint, serial, bigserial, date, timestamp, timestamptz).
After enabling this feature for a column using the `enable_column_stats` command, Timescale will track min/max
values for that column and use that information to exclude chunks for queries that filter by that column and would
not find any data on those chunks. 
* Improved upsert performance by using index scans to verify constraints during inserts on compressed chunks.
This improvement can speed up some ON CONFLICT by more than 100x.
* Improved performance of updates, deletes on all types of compressed chunks, as well as inserts into compressed
chunks with unique constraints by filtering data while accessing the compressed data and before decompressing.
This improvement can speedup significantly a lot of types of DML command by signaling constraint violations without decompressing or decompressing only when matching records are found in the case of updates, deletes and upserts. 
* New ability to add Foreign Keys from regular (non hyper) tables to hypertables, with all types of cascading options supported. 
Really useful for hypertables that partition using sequntial IDs, but also for some time series use cases. 
* Advanced handling of Foreign Keys on compressed chunks do eliminate locking of referenced tables when new 
chunks are compressed. No more DML getting blocked on referenced tables while compression runs on a hypertable.
* Improved support for queries on Continuous Aggregates - more types of joins supported (INNER/LEFT and LATERAL joins),
more than one equality operators on join clauses allowed and support for joining with multiple regular tables.

**PostgreSQL 13 support removal announcement**
Following the deprecation announcement for PostgreSQL 13 in TimescaleDB 2.13,
PostgreSQL 13 is not supported starting with TimescaleDB 2.16.
Currently supported PostgreSQL major versions are 14, 15 and 16.
```

Reply:
@svenklemm @antekresic @fabriziomello can you check and let me know if this looks good & whether I missed anything?

@pallavisontakke I am missing more than a few things that are in the [milestone](https://github.com/timescale/timescaledb/milestone/86) from the description and the list that follows. Like for example the PR by Fabrizio on the CAggs improvements and various other PRs above 7100

@sven check what I wrote about FKs - I am assuming that this will make it for the release. Otherwise, we should remove

Reply:
@billy-the-fish we are getting close to 2.16 being ready - do you want to check my proposal above so that we are prepared for when we'll go full speed to start the final PR and tag?

Reply:
Hi Yannis, I've updated CHANGELOG.md & PR-conversation with your feature-description suggestions. That may help others to review it in the current/expected state.

As this branch was created a few days back (for a tentative PR), it does not include the other release-notes. However, they would be there in the final PR.